migrations/1_check_authority_address.js

```diff
@@ -1,40 +1,68 @@
 require('dotenv').config()
-var RootChain = artifacts.require('./RootChain.sol')
+const RootChain = artifacts.require('./RootChain.sol')
+
 
 module.exports = async function (deployer, network, accounts) {
-  if (process.env.USE_EXISTING_AUTHORITY_ADDRESS) {
+  // Get the deployer address
+  const deployerAddress = (process.env.DEPLOYER_PASSPHRASE) ?
+    await RootChain.web3.eth.personal.newAccount(process.env.DEPLOYER_PASSPHRASE) :
+    (process.env.DEPLOYER_ADDRESS || accounts[0])
+
```

Comment:
How about setting the DEPLOYER_ADDRESS and AUTHORITY_ADDRESS here, so that you don't have to remember to do the ` || accounts[0]` everywhere?
e.g.
`process.env.DEPLOYER_ADDRESS = deployerAddress`

Reply:
Imo, environment variables are suppose to be set inside the environment, not the code.

As the migration process gets more complex, constantly mutating `process.env` will make it harder to debug in the long run

Reply:
Yeah, it's common enough practice but I agree that it can be confusing to modify the env.

Hmm, I still think that always having to remember that the DEPLOYER_ADDRESS (and AUTHORITY_ADDRESS) can be set either via `process.env.DEPLOYER_ADDRESS` or `process.env.DEPLOYER_PASSPHRASE` or in `accounts[0]` is kinda clunky and error-prone though. 
How about setting them as global properties instead?

```
global.deployerAddress = (process.env.DEPLOYER_PASSPHRASE) ?
    await RootChain.web3.eth.personal.newAccount(process.env.DEPLOYER_PASSPHRASE) :
    (process.env.DEPLOYER_ADDRESS || accounts[0])
```

Reply:
Yes, its definitely clunky and `global.<x>` would be a nice compromise 👍